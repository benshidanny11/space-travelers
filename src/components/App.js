import '../assets/styles/main.css';
import { Routes, Route } from 'react-router-dom';
import Head from './Head';
import Rockets from './pages/Rockets';
import Missions from './pages/Missions';
import MyProfile from './pages/MyProfile';

function App() {
  return (
    <div>
      <Head />
      <Routes>
        <Route exact="true" path="/" element={<Rockets />} />
        <Route exact="true" path="/missions" element={<Missions />} />
        <Route exact="true" path="/profile" element={<MyProfile />} />
      </Routes>
    </div>
  );
}

export default App;
