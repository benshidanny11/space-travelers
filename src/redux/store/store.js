import { combineReducers, createStore, applyMiddleware } from 'redux';
import logger from 'redux-logger';
import thunk from 'redux-thunk';

import { rocketsReducer } from '../reducers/rocket';
import { missionsReducer } from '../reducers/mission';

const reducers = combineReducers({
  rockets: rocketsReducer,
  missions: missionsReducer,
});

const store = createStore(
  reducers,
  applyMiddleware(thunk, logger),
);

export default store;
