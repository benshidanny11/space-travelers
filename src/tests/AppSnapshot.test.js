import React from 'react';
import Renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/store/store';
import App from '../components/App';

test('renders learn react link', () => {
  const tree = Renderer
    .create(
      <Provider store={store}>
        <App />
      </Provider>,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
