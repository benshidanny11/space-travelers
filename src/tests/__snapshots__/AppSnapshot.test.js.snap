// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<div>
  <nav>
    <div
      id="logo"
    >
      <img
        alt="App logo"
        src="logo.png"
      />
      <h1>
        Space Travellers' Hub
      </h1>
    </div>
    <div
      className="links"
    >
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
      >
        Rockets
      </a>
      <a
        className=""
        href="/missions"
        onClick={[Function]}
      >
        Mission
      </a>
      <a
        className=""
        href="/profile"
        onClick={[Function]}
      >
        My profile
      </a>
    </div>
  </nav>
  <div />
</div>
`;
